actors/miner/src/lib.rs

```diff
@@ -1950,6 +1957,216 @@ impl Actor {
         Ok(ProveCommitSectors3Return { activation_results: result })
     }
 
+    fn prove_commit_sectors_ni(
+        rt: &impl Runtime,
+        params: ProveCommitSectorsNIParams,
+    ) -> Result<ProveCommitSectorsNIReturn, ActorError> {
+        let policy = rt.policy();
+        let curr_epoch = rt.curr_epoch();
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let info = get_miner_info(rt.store(), &state)?;
+
+        validate_seal_aggregate_proof(
+            &params.aggregate_proof,
+            params.sectors.len() as u64,
+            policy,
+            true,
+        )?;
+
+        rt.validate_immediate_caller_is(
+            info.control_addresses.iter().chain(&[info.worker, info.owner]),
+        )?;
+
+        if params.proving_deadline >= policy.wpost_period_deadlines {
+            return Err(actor_error!(
+                illegal_argument,
+                "proving deadline index {} invalid",
+                params.proving_deadline
+            ));
+        }
+
+        if !deadline_is_mutable(
+            policy,
+            state.current_proving_period_start(policy, curr_epoch),
+            params.proving_deadline,
+            curr_epoch,
+        ) {
+            return Err(actor_error!(
+                forbidden,
+                "proving deadline {} must not be the current or next deadline ",
+                params.proving_deadline
+            ));
+        }
+
+        if consensus_fault_active(&info, rt.curr_epoch()) {
+            return Err(actor_error!(
+                forbidden,
+                "ProveCommitSectorsNI not allowed during active consensus fault"
+            ));
+        }
+
+        if !can_prove_commit_ni_seal_proof(rt.policy(), params.seal_proof_type) {
+            return Err(actor_error!(
+                illegal_argument,
+                "unsupported seal proof type {}",
+                i64::from(params.seal_proof_type)
+            ));
+        }
+
+        if params.aggregate_proof_type != RegisteredAggregateProof::SnarkPackV2 {
+            return Err(actor_error!(illegal_argument, "aggregate proof type must be SnarkPackV2"));
+        }
+
+        let (validation_batch, proof_inputs) = validate_ni_sectors(
+            rt,
+            &params.sectors,
+            params.seal_proof_type,
+            params.require_activation_success,
+        )?;
+
+        if validation_batch.success_count == 0 {
+            return Err(actor_error!(illegal_argument, "no valid NI commits specified"));
+        }
+        let valid_sectors = validation_batch.successes(&params.sectors);
+
```

Comment:
❗ This is missing a call to `state.allocate_sector_numbers(store, &sector_numbers, CollisionPolicy::DenyCollisions)`. I spotted this by reading through the code for pre-commit during this review. Please do a re-read of that code yourself too to ensure we've included all necessary workflow.